### ISSUE TITLE
Hashing performance degradation when using large blocks

### ISSUE THREAD

**Issue by kpumuk** [syncthing/syncthing#5557]
I have an interesting issue, probably caused by me not understanding something. I have a 20gb file I want to sync between devices. It is static, and no other process is modifying it.

Now, what I see. When “Use Large Blocks” is not enabled, I get consistent 188.5 MiB/s indexing performance:

```
2019-02-22 15:43:51 Walk 5rslf-uzcoq [] current progress 1964900352/23638202528 at 188.5 MiB/s (8%)
```

When I enable “Use Large Blocks”, hashing only shows 3.4 MiB/s:

```
2019-02-22 15:38:52 Walk 5rslf-uzcoq [] current progress 167772160/23638202528 at 3.4 MiB/s (0%)
```

On startup, benchmark shows:

```
2019-02-22 11:14:05 Single thread SHA256 performance is 497 MB/s using minio/sha256-simd (94 MB/s using crypto/sha256).
2019-02-22 11:14:06 Hashing performance is 205.38 MB/s
```

All other settings for the folder have default values, installed version is 1.0.1. What am I doing wrong? Thank you

### Version Information

Syncthing Version: v1.0.1, v1.1.0-rc.1

**Comment by kpumuk** [comment]
Supposed fix: https://github.com/syncthing/syncthing/pull/5556

**Comment by calmh** [comment]
Fixed in #5556 (thanks for reporting!)
